src/analysis/normed_space/inner_product.lean

```diff
@@ -1374,6 +1382,61 @@ linear_map.mk_continuous
 
 end norm
 
+section bessels_inequality
+
+variables {ι: Type*} (x : E) {v : ι → E}
+
+/-- Bessel's inequality for finite sums. -/
+lemma orthonormal.sum_inner_products_le {s : finset ι} (hv : orthonormal 𝕜 v) :
+  ∑ i in s, ∥⟪v i, x⟫∥ ^ 2 ≤ ∥x∥ ^ 2 :=
+begin
+  have h₂ : ∑ i in s, ∑ j in s, ⟪v i, x⟫ * ⟪x, v j⟫ * ⟪v j, v i⟫
+    = (∑ k in s, (⟪v k, x⟫ * ⟪x, v k⟫) : 𝕜),
+   { exact hv.inner_left_right_finset },
+  have h₃ : ∀ z : 𝕜, re (z * conj (z)) = ∥z∥ ^ 2,
+  { intro z,
+    simp only [mul_conj, norm_sq_eq_def'],
+    norm_cast, },
+  rw ←sub_nonneg,
+  suffices hbf: ∥x -  ∑ i in s, ⟪v i, x⟫ • (v i)∥ ^ 2 = ∥x∥ ^ 2 - ∑ i in s, ∥⟪v i, x⟫∥ ^ 2,
+  { rw ←hbf,
```

Comment:
Did you miss this one, or prefer not to implement it?

Reply:
I missed this before, but I'm not actually sure what's being suggested. Sorry! Would you kindly explain again? Thanks.

Reply:
Sorry for the ambiguity.  This is an opportunity to reduce 3 lines to 2 (which always makes us happy!). In this case, you can move the `rw ←sub_nonneg` inside the `suffices` block and the proof goes through without change -- and then, since the `suffices` block  would begin 
```lean
  { rw ←sub_nonneg,
    rw ←hbf,
```
we might as well combine these two lines, to
```lean
 { rw [←sub_nonneg, ←hbf],
```